CI/CD Failure: `Sphinx==9.1.0` requires Python 3.12+

#### Description

The GitHub Actions workflow for generating documentation (GitHub Pages) is failing during the **"Install dependencies"** step. The build logs report that `pip` cannot find a matching distribution for `Sphinx==9.1.0`.

#### Error logs

```text
...
ERROR: Ignored the following yanked versions: 8.3.0
ERROR: Ignored the following versions that require a different python version: 1.21.2 Requires-Python >=3.7,<3.11; 1.21.3 Requires-Python >=3.7,<3.11; 1.21.4 Requires-Python >=3.7,<3.11; 1.21.5 Requires-Python >=3.7,<3.11; 1.21.6 Requires-Python >=3.7,<3.11; 9.1.0 Requires-Python >=3.12; 9.1.0rc1 Requires-Python >=3.12; 9.1.0rc2 Requires-Python >=3.12
ERROR: Could not find a version that satisfies the requirement Sphinx==9.1.0 (from versions: ...
...)
ERROR: No matching distribution found for Sphinx==9.1.0
Error: Process completed with exit code 1.
```

#### Root cause

The `github-pages.yml` workflow was configured to use **Python 3.11**. However, **Sphinx 9.1.0** (and all versions 9.x+) has dropped support for Python 3.11 and now strictly requires **Python 3.12 or higher**.

#### Solution

Update the Python version in the GitHub Actions workflow configuration to satisfy the dependency requirements.

**File:** `.github/workflows/github-pages.yml`

**Change:**
```yaml
# Before
- name: Set up Python
  uses: actions/setup-python@v5
  with:
    python-version: '3.11'
```
```yaml
# After
- name: Set up Python
  uses: actions/setup-python@v5
  with:
    python-version: '3.12'
```

#### **Checklist**
- [ ] Update `python-version` in `github-pages.yml`.
- [ ] Verify if other environment dependencies are compatible with 3.12.

--- a/.github/workflows/github-pages.yml
+++ b/.github/workflows/github-pages.yml
@@ -24,7 +24,7 @@ jobs:
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: '3.11'
+          python-version: '3.12'
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip

